### ISSUE TITLE
Investigate quickly blocking malicious IP addresses on applications

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#6242]
# Story

As a web team, I would like to determine if there is a way we can block multiple IP addresses quickly for applications to help mitigate active attacks.

# Additional Info
https://docs.acquia.com/cloud-platform/arch/security/restrict/ contains info about using .htaccess to block IP addresses but we would like to avoid the existing lengthy processes to deploy these changes. 

Could we pull in a list of IPs from a Acquia Environment variable or something similar?

# Proposed solution

**Comment by joewhitsitt** [comment]
https://docs.acquia.com/cloud-platform/develop/env-variable/#environment-variables-in-htaccess
https://docs.acquia.com/cloud-platform/manage/variables/

I haven't seen an example of an array or a long multiline variable though.

**Comment by pyrello** [comment]
It looks like you can deny a list of IP addresses by using a space between them: 

https://kb.veerotech.net/how-to-block-ips-with-htaccess/#:~:text=want%20to%20block.-,To%20Deny%20Multiple%20IP's%20in%20a%20Single%20Line%20of%20Code,with%20each%20IP%20for%20separation.&text=All%20you%20need%20to%20do,htaccess%20file.

My idea is that we would create a [custom environment variable](https://docs.acquia.com/cloud-platform/manage/variables/) called something like `UIOWA_BLOCKED_IPS` and set the value to a string like `193.42.33.66 179.128.33.88` (I made up the second IP address). We would use the [`SetEnvIfExpr`](https://httpd.apache.org/docs/2.4/mod/mod_setenvif.html#SetEnvIfExpr) directive to compare our env variable using an [Apache expression](https://httpd.apache.org/docs/2.4/expr.html#examples).

I'm not totally sure how it all fits together yet, but it seems like we have the pieces and enough flexibility that we could build a tool that would allow us to block 1-10 IP addresses via .htaccess for the short term.

**Comment by pyrello** [comment]
It might actually be as simple as just checking for the presence of the env variable, then having the Deny directive set to the env variable.

https://stackoverflow.com/a/53650767/1375741

**Comment by cory-skeers** [comment]
> It looks like you can deny a list of IP addresses by using a space between them:

post-grooming convo: Spaces aren't allowed in environmental variables (and there's also a pretty small limit on variable size), but there may be a way to set up the expression (use `|` or `,`, replace with spaces in the expression) to still accomplish this.

**Comment by joewhitsitt** [comment]
FWIW, that ip hasn't registered an event in splunk since 3/7 at 1pm.

**Comment by briand44** [comment]
Yeah, I am thinking we reframe this issue to be an investigation into how we can quickly block multiple IP's in the case of an attack for both D7 and D9 sites and regroom and reprioritize this.

**Comment by joewhitsitt** [comment]
https://www.drupal.org/docs/8/core/modules/ban/overview could be used for d9 sites as documented on https://docs.acquia.com/cloud-platform/arch/security/restrict/

Most likely this would allow multiple addresses as default config that could be deployed or set with blt ume (ignored config), but it would be at the bootstrapped layer of drupal, not htaccess.

**Comment by joewhitsitt** [comment]
I had some recent thoughts about IPs/Spam captured in a team card (placed in backlog near-term):

> Noticed our favorite webform submission expert, David Lee, is mainly sending from a single IP address (Remote IP address: 1.163.240.100). Could we just look into adding a shared IP ban list? Perhaps as a... config entity?!
> 
> If not shared, there is the core `Ban` module that we could add to all sites as just one more tool at our disposal to help sites on a case by case basis. Might have been an easier way to ban that one DoS attacker a while back. Drupal still bootstraps but probably of a hit overall (403 response). Also prevents banning your own IP. Maybe we can make sure it doesn't ban the campus ranges so we can still get in via VPN?

**Comment by joewhitsitt** [comment]
Trying to find the Acquia issue/thread related to the noise in NewRelic's error inbox. I think I recall they didn't have a fix for it yet but I think it is worth checking in on.

Traffic will still get to htaccess but not fully bootstrap drupal if we purposely disallow junk paths we know they hit that we don't serve and this will reduce a lot of the noise.

```
# Block known scanner paths — return 403 before Drupal
RewriteRule ^owa/ - [F,L]
RewriteRule ^autodiscover/ - [F,L]
RewriteRule ^wp-login\.php - [F,L]
RewriteRule ^wp-admin/ - [F,L]
RewriteRule ^xmlrpc\.php - [F,L]
RewriteRule ^\.env - [F,L]
```
